src/Lykke.AlgoStore.Api/RealTimeStreaming/DataStreamers/WebSockets/Middleware/WebSocketMiddleware.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Net;
+using System.Threading.Tasks;
+using Lykke.AlgoStore.Api.RealTimeStreaming.DataStreamers.WebSockets.Handlers;
+using Microsoft.AspNetCore.Http;
+
+namespace Lykke.AlgoStore.Api.RealTimeStreaming.DataStreamers.WebSockets.Middleware
+{
+    public class WebSocketMiddleware<T> : WebSocketMiddlewareBase<T>
+    {
+        private readonly RequestDelegate _next;
+
+        public WebSocketMiddleware(RequestDelegate next)
+        {
+            _next = next;
+        }
+
+        public async Task Invoke(HttpContext context, WebSocketHandlerBase<T> webSocketHandler)
```

Comment:
You can allow custom handlers by making the handler a generic constraint inheriting from WebSocketHandlerBase<T>. This way you can avoid having the separate Candles middleware